services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/S3JavaMultipartTransferProgressListenerTest.java

```diff
@@ -350,4 +348,14 @@ void copyWithJavaBasedClient_listeners_reports_ProgressWhenSuccess_copy() throws
         int numTimesBytesTransferred = 2;
         Mockito.verify(transferListenerMock, times(numTimesBytesTransferred)).bytesTransferred(ArgumentMatchers.any());
     }
+
+
+    private static void assertTransferListenerCompletion(CaptureTransferListener transferListener) {
+        assertTimeoutPreemptively(
```

Comment:
Could you add a small message to `assertTimeoutPreemptively` (there is an overload method that accepts one) that mentions the transfer didn't finish within 5 seconds or something like that?